qutip/quantum_systems/jaynes_cummings.py

```diff
@@ -0,0 +1,192 @@
+    if rotating_wave:
+        # Rotating wave approximation: g(a†σ_- + aσ_+)
+        H_interaction = g * (operators['a_dag'] * operators['sigma_minus'] +
+                             operators['a'] * operators['sigma_plus'])
+    else:
+        # Full interaction (Rabi model): g(a† + a)(σ_+ + σ_-)
+        H_interaction = g * (operators['a_dag'] + operators['a']) * \
+            (operators['sigma_plus'] + operators['sigma_minus'])
+
+    # Total Hamiltonian
+    hamiltonian = H_cavity + H_atom + H_interaction
+
+    # Build collapse operators for dissipation
+    c_ops = []
+
+    # Cavity relaxation: kappa(1 + n_th) * a
+    cavity_relax_rate = cavity_decay * (1 + thermal_photons)
+    if cavity_relax_rate > 0.0:
+        c_ops.append(np.sqrt(cavity_relax_rate) * operators['a'])
+
+    # Cavity excitation (thermal): kappa * n_th * a_dag
+    cavity_excite_rate = cavity_decay * thermal_photons
+    if cavity_excite_rate > 0.0:
+        c_ops.append(np.sqrt(cavity_excite_rate) * operators['a_dag'])
+
+    # Atomic spontaneous emission: gamma * sigma_minus
+    if atomic_decay > 0.0:
+        c_ops.append(np.sqrt(atomic_decay) * operators['sigma_minus'])
+
+    # Atomic pure dephasing: gamma_phi * sigma_z
+    if atomic_dephasing > 0.0:
+        c_ops.append(np.sqrt(atomic_dephasing) * operators['sigma_z'])
+
+    # LaTeX representation
+    if rotating_wave:
+        latex = (r"H = \omega_c a^\dagger a + \frac{\omega_a}{2}\sigma_z + "
+                 r"g(a^\dagger\sigma_- + a\sigma_+)")
+    else:
+        latex = (r"H = \omega_c a^\dagger a + \frac{\omega_a}{2}\sigma_z + "
+                 r"g(a^\dagger + a)(\sigma_+ + \sigma_-)")
+
+    # Set all attributes in the system
+    system.operators = operators
+    system.hamiltonian = hamiltonian
+    system.c_ops = c_ops
+    system.latex = latex
+
+    return system
+
+
+# Example usage and demonstrations
+if __name__ == "__main__":
+    print("Jaynes-Cummings Model Examples")
+    print("=" * 40)
+
+    # Example 1: Basic resonant system
+    print("\n1. Basic Resonant Jaynes-Cummings System:")
+    jc_basic = jaynes_cummings(omega_c=1.0, omega_a=1.0, g=0.1, n_cavity=5)
+    jc_basic.pretty_print()
+
+    # Example 2: Detuned system with dissipation
+    print("\n2. Detuned System with Dissipation:")
+    jc_dissipative = jaynes_cummings(
+        omega_c=1.0,
+        omega_a=1.1,  # 10% detuning
+        g=0.05,
+        n_cavity=8,
+        cavity_decay=0.01,    # kappa = 0.01
+        atomic_decay=0.005,   # gamma = 0.005
+        thermal_photons=0.1   # n_th = 0.1
+    )
+    jc_dissipative.pretty_print()
+
+    # Example 3: Rabi model (no rotating wave approximation)
+    print("\n3. Rabi Model (No RWA):")
+    rabi = jaynes_cummings(
+        omega_c=1.0,
+        omega_a=1.0,
+        g=0.2,
+        rotating_wave=False
+    )
+    print(f"LaTeX: {rabi.latex}")
+
+    # Example 4: Accessing operators and Hamiltonian
+    print("\n4. Accessing System Components:")
```

Comment:
when this is finished/moved to a test or notebook (or docs), would be nice to show how to feed these too mesolve. hopefully at that point send the whole quantumsystem will work too of course